static/app/views/replays/detail/console/useConsoleFilters.tsx

```diff
@@ -0,0 +1,82 @@
+import {useCallback, useMemo} from 'react';
+import {browserHistory} from 'react-router';
+
+import type {BreadcrumbTypeDefault, Crumb} from 'sentry/types/breadcrumbs';
+import {isBreadcrumbTypeDefault} from 'sentry/types/breadcrumbs';
+import {decodeList, decodeScalar} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+import {filterItems} from 'sentry/views/replays/detail/utils';
+
+type Item = Extract<Crumb, BreadcrumbTypeDefault>;
+
+type Options = {
+  breadcrumbs: Crumb[];
+};
+
+type Return = {
+  items: Item[];
+  logLevel: string[];
+  searchTerm: string;
+  setLogLevel: (logLevel: string[]) => void;
+  setSearchTerm: (searchTerm: string) => void;
+};
+
+const FILTERS = {
+  logLevel: (item: Item, logLevel: string[]) =>
+    logLevel.length === 0 || logLevel.includes(item.level),
+
+  searchTerm: (item: Item, searchTerm: string) =>
+    JSON.stringify(item.data).toLowerCase().includes(searchTerm),
+};
+
+function useConsoleFilters({breadcrumbs}: Options): Return {
+  const {pathname, query} = useLocation();
+
+  const typeDefaultCrumbs = useMemo(
+    () => breadcrumbs.filter(isBreadcrumbTypeDefault),
+    [breadcrumbs]
+  );
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  const stringyLogLevel = JSON.stringify(query.consoleLogLevel);
+  const logLevel = useMemo(() => decodeList(query.consoleLogLevel), [stringyLogLevel]);
+  /* eslint-enable react-hooks/exhaustive-deps */
```

Comment:
Which dep list would this rule yell at?

Reply:
I didn't want to pass in `query.consoleLogLevel` because it's a `string[]`, so it'll be a new object often enough. Instead I was playing with shoving that array into JSON.stringify and then using that as the hook dep.

`useMemo` complains because it expected `query.consoleLogLevel` to be in the deps array.